convert2rhel/actions/__init__.py

```diff
@@ -230,63 +285,199 @@ def get_actions(actions_path, prefix):
     return actions
 
 
-def resolve_action_order(potential_actions, failed_actions):
+class Stage:
+    def __init__(self, stage_name, task_header=None, next_stage=None):
+        self.stage_name = stage_name
+        self.task_header = task_header if task_header else stage_name
+        self.next_stage = next_stage
+        self._has_run = False
+
+        python_package = importlib.import_module("convert2rhel.actions.%s" % self.stage_name)
+        self.actions = get_actions(python_package.__path__, python_package.__name__ + ".")
+
+    def check_dependencies(self, previous_stage_actions=None):
+        """
+        Make sure dependencies of this Stage and previous stages are satisfied.
+
+        :raises DependencyError: when there is an unresolvable dependency in
+            the set of actions.
+        """
+        # We want to throw an exception if one of the actions fails to resolve
+        # its deps.  We don't care about the return value here.
+        actions_so_far = list(resolve_action_order(self.actions))
+
+        if self.next_stage:
+            self.next_stage.check_dependencies(actions_so_far)
+
+    def run(self, successes=None, failures=None, skips=None):
+        """
+        Run all the actions in Stage and other linked Stages.
+
+        :keyword successes: Actions which have already run and succeeded.
+        :type: Sequence
+        :keyword failures: Actions which have already run and failed.
+        :type: Sequence
+        :return: 2-tuple consisting of two lists.  One with Actions that
+            have succeeded and one of Actions that have failed.  These
+            lists contain the Actions passed in via successes and failures
+            in addition to any that were run in this :class`Stage`.
+
+        :rtype: FinishedActions
+        .. note:: Success is currently defined as an action whose status after
+            running is WARNING or better (WARNING or SUCCESS) and
+            failure as worse than WARNING (OVERRIDABLE, ERROR, FATAL)
```

Comment:
What about transforming this into an `.. warning::`? I think this note is a very important thing for people no pay attention to.

Reply:
Oh, that's looks way better! I like the `important` more than the `warning`! Thanks for finding that out.